aws-rds-cfn-common/src/test/java/software/amazon/rds/common/logging/LoggingProxyClientTest.java

```diff
@@ -0,0 +1,93 @@
+package software.amazon.rds.common.logging;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import software.amazon.awssdk.awscore.AwsRequest;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.pagination.sync.SdkIterable;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.rds.common.handler.ProxyClientTestBase;
+import software.amazon.rds.common.printer.FilteredJsonPrinter;
+
+@ExtendWith(MockitoExtension.class)
+class LoggingProxyClientTest extends ProxyClientTestBase {
+    public static final String AWS_ACCOUNT_ID = "123456789";
+    public static final String TOKEN = "token";
+    public static final String STACK_ID = "stackId";
+
+    @Mock
+    private ProxyClient<RdsClient> proxy;
+
+    @Mock
+    private AwsRequest awsRequest;
+
+    @Mock
+    private AwsResponse awsResponse;
+
+    @Mock
+    private SdkIterable<AwsResponse> SdkIterable;
+
+    @Mock
+    private CompletableFuture<AwsResponse> awsResponseCompletableFuture;
+
+    private ResourceHandlerRequest<Void> request;
+
+    @BeforeEach
+    public void setup() {
+        request = new ResourceHandlerRequest<>();
+        request.setAwsAccountId(AWS_ACCOUNT_ID);
+        request.setClientRequestToken(TOKEN);
+        request.setStackId(STACK_ID);
+    }
+
+    private ProxyClient<RdsClient> getProxyRdsClient(Logger logger) {
+        return new LoggingProxyClient<>(new RequestLogger(logger, request, new FilteredJsonPrinter()), proxy);
+    }
+
+    @Test
+    void test_injectCredentialsAndInvokeV2() {
+        Logger logger = m -> assertThat(m.contains(AWS_ACCOUNT_ID)).isTrue();
```

Comment:
We should probably assert that it was invoked.

Reply:
Done.